sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2916,14 +2916,14 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    * }}}
    */
   override def visitAddTableColumns(ctx: AddTableColumnsContext): LogicalPlan = withOrigin(ctx) {
-    AlterTableAddColumnsStatement(
-      visitMultipartIdentifier(ctx.multipartIdentifier),
+    AlterTableAddColumns(
+      UnresolvedTable(visitMultipartIdentifier(ctx.multipartIdentifier)),
```

Comment:
I will be making a change to use `UnresolvedTableOrView` so that we can keep the existing error message like `Cannot alter a view with ALTER TABLE. Please use ALTER VIEW instead`. If not, we will just see `v1 is a view not a table`. @cloud-fan, sounds right?

Reply:
`UnresolvedTableOrView` wouldn't work since it can resolve to a temp view. Do you have any suggestion if we need to provide the hint?

Reply:
I think `v1 is a view not a table` is good enough for now. In the future we can add an error message parameter in `UnresolveTable/View`, so that each command can customize its error message.

Reply:
or we can do it now, so that we don't need to update a lot of tests in this PR.

Reply:
Thanks, I will do it in this PR.

But I wasn't sure if the error message parameter would work well because you can't guarantee the error message is mapped to the actual error condition (unless I misunderstood).

How about we introduce the "actual" resolved plan? For example, in `Analyzer.scala`
```
case u @ UnresolvedTable(identifier) =>
  lookupTableOrView(identifier).map {
    case v: ResolvedView => u.copy(actual = v)
    case table => table
  }.getOrElse(u)
```
Then, in `CheckAnalysis.checkAnalysis`,
```
case AlterTable(u @ UnresolvedTable(_, actual: ResolvedView), _) =>
   u.failAnalysis("Cannot alter a view with ALTER TABLE. Please use ALTER VIEW instead")
```

Reply:
yea that also works.

nit: we should skip relation lookup if the actual plan is already set. Or we can introduce a new node `UnresolvedTableWithViewExists` to explicitly indicate this case.

Reply:
I went with your suggestion since it was simpler. Thanks!